smartpqi/arcconf.go

```diff
@@ -66,12 +60,23 @@ func parseLogicalDeviceString(rawData string) (LogicalDevice, error) {
 	}
 	ld := LogicalDevice{}
 
+	if len(rawData) < 1 {
+		return ld, fmt.Errorf("cannot parse an empty string")
+	}
+
 	// break data into first line which has the logical device number and
 	// the remaining data is in mostly key : value pairs
 	lineData := strings.SplitN(rawData, "\n", 2)
+	if len(lineData) < 2 {
+		return ld, fmt.Errorf("expected more than 2 lines of data, found %d", len(lineData))
```

Comment:
Sorry about the nit-picking here, but technically you are only checking for at least 2 lines of data;  your error message (here and below) says 'more than 2', which could mislead a user.